### PR TITLE
create an authentication server

### PR DESCRIPTION
**Pull request by Jamesxu182**

### CODE REVIEW

**Review by bonellon — DISMISSED** [review]

File: auth/src/main/java/com/wayfinder/auth/config/AuthorisationServerConfig.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
+		clients
+	    	.inMemory()
+		        .withClient("web-app")
+		        .secret(passwordEncoder.encode("123456"))
+		        .autoApprove(true)
+		        .authorizedGrantTypes("implicit", "refresh_token", "password", "authorization_code", "client_credentials")
+		        .accessTokenValiditySeconds(1800)
+		        .scopes("FOO")
+		        .redirectUris("http://www.google.com");
```
Comment (ats0stv) [inline auth/src/main/java/com/wayfinder/auth/config/AuthorisationServerConfig.java]: Please change the redirect URI from google.com to something like localhost or so

File: auth/src/main/java/com/wayfinder/auth/config/SwaggerConfig.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@Configuration
+@EnableSwagger2
+public class SwaggerConfig {
+    @Bean
+    public Docket api() { 
+        return new Docket(DocumentationType.SWAGGER_2)
```
Comment (ats0stv) [inline auth/src/main/java/com/wayfinder/auth/config/SwaggerConfig.java]: Please add our app specific information
```java
@Configuration
@EnableSwagger2
public class SwaggerConfig {

    @Bean
    public Docket api()
    {
        return new Docket(DocumentationType.SWAGGER_2).apiInfo(wayfinderInfo())
                                                      .select()
                                                      .apis(RequestHandlerSelectors.any())
                                                      .paths(PathSelectors.any())
                                                      .build();
    }

    private ApiInfo wayfinderInfo()
    {
        Contact contact2 = new Contact("Wayfinders", "", "thundyia@tcd.ie");
        return new ApiInfoBuilder().title("Trinity Wayfinders Authentication Module")
                                   .description("Authentication ReST API Documentation")
                                   .contact(contact2)
                                   .version("0.1").build();
    }
}
```

File: auth/bin/main/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/com/
```
Comment (ats0stv) [inline auth/bin/main/.gitignore]: This file should be ignored. All of /bin/* should not be commited

File: auth/bin/main/application.properties [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## Spring DATASOURCE (DataSourceAutoConfiguration & DataSourceProperties)
+spring.datasource.url = jdbc:mysql://localhost:3306/wayfinder?useSSL=false
```
Comment (ats0stv) [inline auth/bin/main/application.properties]: This file should be ignored. All of /bin/* should not be committed

**Review by ats0stv — CHANGES_REQUESTED** [review]

**Review by ats0stv — APPROVED** [review]

File: Code/AuthenticationServer/src/main/java/com/wayfinder/auth/config/AuthorisationServerConfig.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+		        .withClient("web-app")
+		        .secret(passwordEncoder.encode("123456"))
+		        .autoApprove(true)
+		        .authorizedGrantTypes("implicit", "refresh_token", "password", "authorization_code", "client_credentials")
+		        .accessTokenValiditySeconds(1800)
+		        .scopes("FOO")
+		        .redirectUris("http://localhost:8080/");
+	}
+	
+
```
Comment (bonellon) [inline Code/AuthenticationServer/src/main/java/com/wayfinder/auth/config/AuthorisationServerConfig.java:54]: WHY ARE THERE 3 NEWLINES?

File: Code/AuthenticationServer/src/main/java/com/wayfinder/auth/config/SwaggerConfig.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import springfox.documentation.builders.PathSelectors;
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.service.ApiInfo;
+import springfox.documentation.service.Contact;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@Configuration
+@EnableSwagger2
```
Comment (bonellon) [inline Code/AuthenticationServer/src/main/java/com/wayfinder/auth/config/SwaggerConfig.java:16]: I THOUGHT WERE USING OPENAPI 3.0???????

File: Code/AuthenticationServer/src/main/java/com/wayfinder/auth/entity/Authority.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package com.wayfinder.auth.entity;
+
+import org.springframework.security.core.GrantedAuthority;
+
+public class Authority implements GrantedAuthority {
+	/**
+	 * 
+	 */
```
Comment (bonellon) [inline Code/AuthenticationServer/src/main/java/com/wayfinder/auth/entity/Authority.java:8]: NICE COMMENT

File: Code/AuthenticationServer/src/main/java/com/wayfinder/auth/controller/UserController.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+@RestController
+@RequestMapping("/api/user")
+public class UserController {	
+	@Autowired
+	private UserService userService;
+	
+	@GetMapping("/me")
+	@PreAuthorize("hasAuthority('ROLE_USER')")
+	public Object getUser() {
+    	OAuth2Authentication oauth2Authentication = (OAuth2Authentication) SecurityContextHolder.getContext().getAuthentication();
```
Comment (bonellon) [inline Code/AuthenticationServer/src/main/java/com/wayfinder/auth/controller/UserController.java:26]: TEST

**Review by bonellon — APPROVED** [review]

**Pull request merged**